static/js/src/advantage/react/components/Subscriptions/ExpiryNotification/ExpiryNotification.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  Notification,
+  NotificationSeverity,
+  PropsWithSpread,
+} from "@canonical/react-components";
+import type { NotificationProps } from "@canonical/react-components";
+import React, { ReactNode } from "react";
+
+import { UserSubscriptionStatuses } from "advantage/api/types";
+
+export enum ExpiryNotificationSize {
+  Small = "small",
+  Large = "large",
+}
+
+export enum StatusKey {
+  IsExpired = "is_expired",
+  IsInGracePeriod = "is_in_grace_period",
+  IsExpiring = "is_expiring",
+}
+
+type Props = PropsWithSpread<
+  {
+    multiple?: boolean;
+    size: ExpiryNotificationSize;
+    statuses: UserSubscriptionStatuses;
+  },
+  NotificationProps
+>;
+
+type Messages = Record<
+  StatusKey,
+  { [K in ExpiryNotificationSize]?: { message: ReactNode; title?: ReactNode } }
+>;
+
+// The expiry status keys in priority order.
+const STATUS_KEYS = [
```

Comment:
What do you think about renaming this variable instead of relying on a comment? Some examples below.
```suggestion
const STATUS_KEYS_BY_PRIORITY = [
```

```suggestion
const ORDERED_STATUS_KEYS
```

Reply:
Good idea!